app/models/goods_nomenclatures/nested_set.rb

```diff
@@ -124,6 +125,24 @@ module NestedSet
           .without_excluded_types
           .overview
       end
+
+      def_column_accessor :leaf
+
+      dataset_module do
+        def with_leaf_column
+          association_join(tree_node: proc { |ds| ds.join_child_sids })
+            .select_all(:goods_nomenclatures)
+            .select_append(:tree_node__number_indents, :tree_node__depth)
+            .select_append((Sequel.function(:count, :tree_node__child_sid) =~ 0).as(:leaf))
+            .group(qualified_columns + %i[tree_node__number_indents tree_node__depth])
+        end
+
+      private
+
+        def qualified_columns(qualifier = model.table_name)
```

Comment:
Feels like something that should be builtin

Reply:
Agreed, couldn't find it in the docs though so :man_shrugging: